supervisor/hyperpod.go

```diff
@@ -77,10 +78,51 @@ func createHyperPod(f factory.Factory, spec *specs.Spec) (*HyperPod, error) {
 	if userPod.Resource.Memory > 0 {
 		mem = userPod.Resource.Memory
 	}
-	vm, err := f.GetVm(cpu, mem)
-	if err != nil {
-		glog.V(1).Infof("%s\n", err.Error())
-		return nil, err
+
+	// if image contains kernel file then use it first instead of default kernel
+	kernel := utils.FirstExistingFile([]string{
+		filepath.Join(spec.Root.Path, "boot/vmlinuz"),
+		filepath.Join(spec.Root.Path, "vmlinuz"),
+		filepath.Join(spec.Root.Path, "kernel"),
+		hypervisor.DefaultKernel,
```

Comment:
please remove hypervisor.DefaultKernel.
use `kernel == ""` insteead of `kernel == hypervisor.DefaultKernel`